vllm/model_executor/models/qwen2_5_vl.py

```diff
@@ -211,6 +224,244 @@ def all_gather_interleave(local_tensor, hidden_size: int, tp_size: int):
     return result_tensor
 
 
+def get_load_balance_assignment(sizes, num_gpus=2):
+    """
+    Generate load balancing assignment and metadata 
+    for distributing data across GPUs.
+    
+    Args:
+        sizes: List or tensor of flatten image sizes
+        num_gpus: Number of GPUs to balance across
+    
+    Returns:
+        shuffle_indices: 
+            Indices to reorder data for balanced loading
+        unshuffle_indices: 
+            Indices to restore original order
+        gpu_sample_counts: 
+            Number of samples assigned to each GPU
+        image_is_in_rank: 
+            Boolean tensor [num_gpus, num_samples] indicating assignment
+        grouped_sizes_per_gpu: 
+            Total size assigned to each GPU
+    """
+    # Convert to list for efficient indexing and iteration
+    if hasattr(sizes, 'tolist'):  # Handle tensor input
+        sizes = sizes.tolist()
+
+    n_samples = len(sizes)
+
+    # Handle edge cases
+    if n_samples == 0:
+        empty_assignment = torch.zeros((num_gpus, 0), dtype=torch.bool)
+        return [], [], [0] * num_gpus, empty_assignment, [0] * num_gpus
+
+    if n_samples < num_gpus:
+        shuffle_indices = list(range(n_samples))
+        unshuffle_indices = list(range(n_samples))
+        gpu_sample_counts = [1] * n_samples + [0] * (num_gpus - n_samples)
+
+        # Create assignment matrix
+        image_is_in_rank = torch.zeros((num_gpus, n_samples), dtype=torch.bool)
+        grouped_sizes_per_gpu = []
+
+        for gpu_id in range(num_gpus):
+            if gpu_id < n_samples:
+                image_is_in_rank[gpu_id, gpu_id] = True
+                grouped_sizes_per_gpu.append(sizes[gpu_id])
+            else:
+                grouped_sizes_per_gpu.append(0)
+
+        return (shuffle_indices, unshuffle_indices, gpu_sample_counts,
+                image_is_in_rank, grouped_sizes_per_gpu)
+
+    # Phase 1: Ensure minimum samples per GPU
+    # Sort indices by size (small to large for initial assignment)
+    small_to_large_indices = sorted(range(n_samples), key=lambda i: sizes[i])
+
+    gpu_assignments = [[] for _ in range(num_gpus)]
+    gpu_loads = [0.0] * num_gpus
+    used_indices = set()
+
+    # Assign 1 sample to each GPU (starting with smallest)
+    for gpu_id in range(num_gpus):
+        for idx in small_to_large_indices:
+            if idx not in used_indices:
+                gpu_assignments[gpu_id].append(idx)
+                gpu_loads[gpu_id] += sizes[idx]
+                used_indices.add(idx)
+                break
+
+    # Phase 2: Distribute remaining samples using greedy load balancing
+    # Sort remaining indices by size (large to small for better balancing)
+    remaining_indices = [
+        idx for idx in range(n_samples) if idx not in used_indices
+    ]
+    remaining_indices.sort(key=lambda i: sizes[i], reverse=True)
```

Comment:
```suggestion
    remaining_indices = sorted(
        (idx for idx in range(n_samples) if idx not in used_indices),
        key=lambda i: sizes[i],
        reverse=True,
    )
```
Avoid iterating twice

Reply:
Done